llarp/net/net_int.cpp

```diff
@@ -128,6 +128,17 @@ namespace llarp
       return "";
     return tmp;
   }
+
+  template <>
+  std::string
+  nuint128_t::ToString() const
+  {
+    char tmp[INET6_ADDRSTRLEN] = {0};
+    if (!inet_ntop(AF_INET6, (void*)&n, tmp, sizeof(tmp)))
+      return "";
+    return tmp;
+  }
```

Comment:
This feels wrong.  If nuint128_t is only supposed to be an IPv6 address, then it is misnamed; and if it is meant to be a more generic network-order-128-bit-int then ToString is misnamed (and should rather be something like `ToIPv6String`).

(The same applies to the existing [nh]uint32_t and IPv4 address stringification).

Reply:
huint types need to be made invisible or internal and everything really should be in network byte order, the reason this pr took so long is because i actually did make it do that but ended up reverting it as it became.... huge.

the better names for `nuint16_t` `nuint32_t` and `nuint128_t` would be something like `net_port_t` `ip4_address_t` and `ip6_address_t` respectively, i supposed i will do this next time i refactor all the endian related parts.